### PR TITLE
 add prop types & document

### PR DESCRIPTION
**Pull request by zarsky-broad**
The primary reason for this is to make components consistent and easier to use for newer developers. The potential error catching is a bonus.

### CODE REVIEW

File: src/components/ErrorView.js [hunk 1]
```diff
@@ -39,6 +40,12 @@ const jsonErrorView = ({ statusCode, source, causes, stackTrace, message, except
...
 
 export default class ErrorView extends Component {
+  static propTypes = {
+    collapses: PropTypes.bool,
+    containerStyle: PropTypes.object,
+    error: PropTypes.any.isRequired
+  }
+
   render() {
     const { collapses=true, containerStyle, error } = this.props
```
Comment (katevoss) [inline src/components/ErrorView.js]: Would it be better to make this into a separate block?
Something like:
```
static defaultProps = {
    collapses: true
  }
```

File: src/components/Modal.js [hunk 2]
```diff
@@ -27,19 +29,26 @@ const styles = {
...
 export default class Modal extends Component {
+  static defaultProps = {
+    showCancel: true,
+    showX: false,
+    width: 450
+  }
+
+  static propTypes = {
+    children: PropTypes.node,
+    okButton: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.any]),
```
Comment (MattPutnam) [inline src/components/Modal.js]: Should the last thing be PropTypes.node?

**Review by MattPutnam — COMMENTED** [review]
I still don't like `children` as a prop. It makes it seem like you should pass it like any other prop.

I don't like how they're alphabetical, I'd prefer to see them in a semantic order.

File: src/components/ErrorView.js (hunk 1 above)
Comment (zarsky-broad) [reply src/components/ErrorView.js]: Yup.

File: src/components/Modal.js (hunk 2 above)
Comment (zarsky-broad) [reply src/components/Modal.js]: Yup.

**Review by panentheos — APPROVED** [review]
Looks good.

**Pull request merged**